### PR TITLE
[BuildSystem] Add pipeline configuration to the repository

### PR DESCRIPTION
**Pull request by emlynmac**
## Purpose
Migrating the CI process to be managed through the repository.
This is the first step; enabling our CI pipeline.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [x] Other... Please describe: Add CI pipeline to the repository

## Pull request checklist

This PR has considered:
- [ ] Color Theming
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] View Data Injection
- [ ] Demo App UI/UX update

## How to Test
No code changes, check new build job is fired

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# Performs basic testing for PRs and pushes to specific branches
+
+# Run on pushes to these branches
+trigger:
+  branches:
+    include:
+      - main
+      - develop
+      - release/*
+      - feature/*
```
Comment (JoshuaLai) [inline azure-pipelines.yml:11]: If we are running it on every feature branch should we also run them on `refactor/*` or `bug/*` `defect/*` branches? 
Part of me thinks no cause all that will be triggered again in the develop branch anyways? Is feature in here explicitly for the chat feature branch?

Comment (emlynmac) [reply azure-pipelines.yml:11]: We can if we want. It's not difficult to change so maybe we start with these?

**Review by jimchou-dev — COMMENTED** [review]
Does it make sense to have the file in the root directory of the repo? Maybe move it to some folder under /scripts for organization?
Also probably can make the filename more specific to the job it's doing as I can see we will have other azure pipelines in the future

**Comment by JoshuaLai** [comment]
> Does it make sense to have the file in the root directory of the repo? Maybe move it to some folder under /scripts for organization? Also probably can make the filename more specific to the job it's doing as I can see we will have other azure pipelines in the future

I agree with this, but we may also want to organize the scripts folder a bit more. I think right now we just have the script to do the version updates, maybe inside scripts we can have a new folder for pipelines, and drop this in there?

**Comment by emlynmac** [comment]
> Does it make sense to have the file in the root directory of the repo? Maybe move it to some folder under /scripts for organization?
> Also probably can make the filename more specific to the job it's doing as I can see we will have other azure pipelines in the future

I expect we'll have only one automated CI pipeline file. 

`scripts` does seem like a natural home for this, but the more standard practice in azure is to have an `eng` folder at the top level and then contain things there. 

eg.
 https://github.com/Azure/communication-ui-library-android/tree/develop/eng
 https://github.com/Azure/azure-sdk-for-ios/tree/main/eng
 
 I'll move the file there for now, and create a task for migrating the other files

**Review by vhuseinova-msft — DISMISSED** [review]

**Review by vhuseinova-msft — APPROVED** [review]

**Review by jimchou-dev — APPROVED** [review]

**Pull request merged**